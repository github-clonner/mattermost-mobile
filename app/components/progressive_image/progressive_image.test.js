// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from '@mm-redux/constants/preferences';

import ProgressiveImage from './progressive_image';

jest.mock('react-native-fast-image', () => ({
    preload: jest.fn(),
}));

jest.useFakeTimers();

describe('ProgressiveImage', () => {
    test('should match snapshot for Image', () => {
        const baseProps = {
            isBackgroundImage: false,
            imageUri: 'https://images.com/image.png',
            onError: jest.fn(),
            resizeMethod: 'auto',
            resizeMode: 'contain',
            theme: Preferences.THEMES.default,
            tintDefaultSource: false,
        };

        const wrapper = shallow(<ProgressiveImage {...baseProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for Default Image', () => {
        const baseProps = {
            isBackgroundImage: false,
            defaultSource: 'https://images.com/image.png',
            onError: jest.fn(),
            resizeMethod: 'auto',
            resizeMode: 'contain',
            theme: Preferences.THEMES.default,
            tintDefaultSource: false,
        };

        const wrapper = shallow(<ProgressiveImage {...baseProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for BackgroundImage', () => {
        const baseProps = {
            isBackgroundImage: true,
            imageUri: 'https://images.com/image.png',
            onError: jest.fn(),
            resizeMethod: 'auto',
            resizeMode: 'contain',
            theme: Preferences.THEMES.default,
            tintDefaultSource: false,
        };

        const wrapper = shallow(<ProgressiveImage {...baseProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});