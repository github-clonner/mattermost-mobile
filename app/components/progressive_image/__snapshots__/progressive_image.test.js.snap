// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressiveImage should match snapshot for BackgroundImage 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <ImageBackground
    resizeMode="cover"
    source={
      Object {
        "uri": "https://images.com/image.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`ProgressiveImage should match snapshot for Default Image 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    onError={[MockFunction]}
    resizeMethod="auto"
    resizeMode="contain"
    source="https://images.com/image.png"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
        null,
      ]
    }
  />
</View>
`;

exports[`ProgressiveImage should match snapshot for Image 1`] = `
<ForwardRef(AnimatedComponentWrapper)
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
      Object {
        "backgroundColor": "rgba(61,60,64,NaN)",
      },
    ]
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    onError={[MockFunction]}
    resizeMethod="auto"
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../dist/assets/images/thumb.png",
      }
    }
    style={
      Array [
        undefined,
        Object {
          "opacity": 0.5,
          "tintColor": "#3d3c40",
        },
      ]
    }
  />
  <ForwardRef(AnimatedComponentWrapper)
    onError={[MockFunction]}
    onLoadEnd={[Function]}
    resizeMethod="auto"
    resizeMode="contain"
    source={
      Object {
        "uri": "https://images.com/image.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
        Object {
          "opacity": 0,
        },
      ]
    }
  />
</ForwardRef(AnimatedComponentWrapper)>
`;
